inf-ruby.el

```diff
@@ -505,10 +506,138 @@ Must not contain ruby meta characters.")
     (comint-send-string (inf-ruby-proc) (concat "\n" term "\n"))
     (when print (ruby-print-result))))
 
+(defface inf-ruby-result-overlay-face
+  '((((class color) (background light)) :background "black" :box (:line-width -1 :color "yellow")))
+  "Face used to display evaluation results at the end of line.")
+
+;; Overlay
+
+(defun inf-ruby--make-overlay (l r type &rest props)
+  "Place an overlay between L and R and return it.
+TYPE is a symbol put on the overlay's category property.  It is
+used to easily remove all overlays from a region with:
+    (remove-overlays start end 'category TYPE)
+PROPS is a plist of properties and values to add to the overlay."
+  (let ((o (make-overlay l (or r l) (current-buffer))))
+    (overlay-put o 'category type)
+    (overlay-put o 'inf-ruby-temporary t)
+    (while props (overlay-put o (pop props) (pop props)))
+    (push #'inf-ruby--delete-overlay (overlay-get o 'modification-hooks))
+    o))
+
+(defun inf-ruby--delete-overlay (ov &rest _)
+  "Safely delete overlay OV.
+Never throws errors, and can be used in an overlay's
+modification-hooks."
+  (ignore-errors (delete-overlay ov)))
+
+(cl-defun inf-ruby--make-result-overlay (value &rest props &key where duration (type 'result)
```

Comment:
Since there is only one caller there, we could move all the default values to the `let` form below, change `cl-defun` to `defun` and drop the `cl-lib` dependency. Nothing against it per se, but it's not really used here. And fewer dependencies is always better.

Reply:
Although this way we'd use the nice docstring descriptions.

Perhaps removing the unused arguments could be done in a separate commit, so they're easy to restore later, in case we'd ever manage to make use of them. I can do this change later myself, if you like.

Reply:
Humm it makes sense to me thank you for this heads up I've removed the dependency and the optional arguments on the following commit https://github.com/nonsequitur/inf-ruby/pull/153/commits/3cdf30dc737a828f956564cfbe38b22acba46bab